### PR TITLE
decorate non uniform pointers when possible

### PR DESCRIPTION
**Pull request by rjodinchr**

### CODE REVIEW

**Comment by rjodinchr** [comment]
`clspv` implementation is breaking something. Most probably in `selectFromSameObject` that is now used more when enabling this feature.

**Comment by rjodinchr** [comment]
https://github.com/google/clspv/pull/967

File: src/device.cpp [hunk 1]
```diff
@@ -845,9 +845,7 @@ bool cvk_device::supports_capability(spv::Capability capability) const {
     case spv::CapabilityVulkanMemoryModel:
         return m_features_vulkan_memory_model.vulkanMemoryModel;
     case spv::CapabilityShaderNonUniform:
-        return m_properties.apiVersion >= VK_MAKE_VERSION(1, 2, 0) ||
-               is_vulkan_extension_enabled(
-                   VK_EXT_DESCRIPTOR_INDEXING_EXTENSION_NAME);
+        return supports_non_uniform_pointers();
```
Comment (kpet) [inline src/device.cpp:848]: They're not really non-uniform "pointers". Maybe `supports_non_uniform_decoration`?

**Review by kpet — CHANGES_REQUESTED** [review]
One name change please. Also this conflicts with https://github.com/kpet/clvk/pull/475 I've just merged, you'll need to rebase.

File: src/device.cpp (hunk 1 above)
Comment (rjodinchr) [reply src/device.cpp]: Done

**Review by kpet — APPROVED** [review]
Thanks!

File: src/device.hpp [hunk 2]
```diff
@@ -270,6 +270,12 @@ struct cvk_device : public _cl_device_id,
 
     bool supports_subgroups() const { return m_has_subgroups_support; }
 
+    bool supports_non_uniform_decoration() const {
+        return m_properties.apiVersion >= VK_MAKE_VERSION(1, 2, 0) ||
+               is_vulkan_extension_enabled(
+                   VK_EXT_DESCRIPTOR_INDEXING_EXTENSION_NAME);
+    }
+
```
Comment (kpet) [inline src/device.hpp:278]:
```suggestion

```

**Pull request merged**